.ado/jobs/universal.yml

```diff
@@ -5,143 +5,218 @@ parameters:
-          searchFolder: $(Build.SourcesDirectory)/vnext/target/$(BuildPlatform)/$(BuildConfiguration)
-          runTestsInIsolation: true
-          platform: $(BuildPlatform)
-          configuration: $(BuildConfiguration)
-          publishRunAttachments: true
-          codeCoverageEnabled: true
-          collectDumpOn: onAbortOnly
-          vsTestVersion: latest
-        condition: and(succeeded(), not(startsWith(variables.BuildPlatform, 'arm')))
+            steps:
+              - checkout: self
+                clean: false
+                submodules: false
 
-      - task: VSTest@2
-        displayName: Run Universal Unit Tests (NetCore)
-        timeoutInMinutes: 5 # Set smaller timeout , due to hangs
-        inputs:
-          testSelector: testAssemblies
-          testAssemblyVer2: |
-            Microsoft.ReactNative.Managed.CodeGen.UnitTests/Microsoft.ReactNative.Managed.CodeGen.UnitTests.dll
-          searchFolder: $(Build.SourcesDirectory)/vnext/target/$(BuildPlatform)/$(BuildConfiguration)
-          runTestsInIsolation: true
-          platform: $(BuildPlatform)
-          configuration: $(BuildConfiguration)
-          publishRunAttachments: true
-          codeCoverageEnabled: true
-          collectDumpOn: onAbortOnly
-          vsTestVersion: latest
-        condition: and(succeeded(), eq(variables.BuildPlatform, 'x64'))
+              - task: PowerShell@2
+                displayName: "Check if this environment meets the development dependencies"
+                inputs:
+                  targetType: filePath
+                  filePath: $(Build.SourcesDirectory)\vnext\Scripts\rnw-dependencies.ps1
+                  arguments: -NoPrompt -Tags buildLab
+                  
+              - template: ../templates/prepare-env.yml
+                parameters:
+                  yarnBuildCmd: build
 
-      - template: ../templates/publish-build-artifacts-for-nuget.yml
-        parameters:
-          artifactName: ReactWindows
-          layoutHeaders: eq('true', variables.LayoutHeaders)
-          contents: |
-            Microsoft.ReactNative\**
-            Microsoft.ReactNative.Managed\**
-            Microsoft.ReactNative.Managed.CodeGen\**
-      
-      - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
-        displayName: 'Component Governance Detection'
-        inputs:
-          alertWarningLevel: Medium
-          scanType: 'Register'
\ No newline at end of file
+              - task: NuGetCommand@2
+                displayName: NuGet restore
+                inputs:
+                  command: restore
+                  restoreSolution: vnext/Microsoft.ReactNative.sln
+                  feedsToUse: config
+                  nugetConfigPath: $(Build.SourcesDirectory)/vnext/NuGet.config
+                  restoreDirectory: packages/
+                  verbosityRestore: Detailed # Options: quiet, normal, detailed
```

Comment:
this shouldn't be necessary anymore as msbuild now has support for restoring from packages.config (pass `/restore /p:RestorePackagesConfig=true` to msbuild)

Reply:
Agree, @NickGerleman Has been experimenting with that. Not part of this PR.

Reply:
will save it for a future to keep PR's single-focused.